scripts/tests/test_validate_generated_site.py

```
@@ -0,0 +1,191 @@
+        for reference, diagnostic in (
+            ("missing.html", "missing local target"),
+            ("index.html#missing", "missing fragment"),
+        ):
+            with self.subTest(reference=reference):
+                site = self._valid_site()
+                (site / "index.html").write_text(
+                    (site / "index.html").read_text(encoding="utf-8") + f'<a href="{reference}">broken</a>',
+                    encoding="utf-8",
+                )
+
+                with self.assertRaisesRegex(validate_generated_site.GeneratedSiteError, diagnostic):
```

Ruff (0.16.1)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)